synapse/federation/transport/server.py

```diff
@@ -1376,6 +1380,40 @@ async def on_PUT(self, origin, content, query, group_id):
         return 200, new_content
 
 
+class FederationSpaceSummaryServlet(BaseFederationServlet):
+    PREFIX = FEDERATION_UNSTABLE_PREFIX + "/org.matrix.msc2946"
+    PATH = "/spaces/(?P<room_id>[^/]*)"
+
+    async def on_POST(
+        self,
+        origin: str,
+        content: JsonDict,
+        query: Mapping[bytes, Sequence[bytes]],
+        room_id: str,
+    ) -> Tuple[int, JsonDict]:
+        suggested_only = content.get("suggested_only", False)
+        if not isinstance(suggested_only, bool):
+            raise SynapseError(
+                400, "'suggested_only' must be a boolean", Codes.BAD_JSON
```

Comment:
Should this (and below) be `INVALID_PARAM` instead of `BAD_JSON`?

Reply:
I *think* `M_INVALID_PARAM` is for query-string params, though this is very unclear, and I've opened https://github.com/matrix-org/matrix-doc/issues/3070 about it.

Reply:
I think we're inconsistent with this in Synapse then!

Reply:
s/synapse/the entire matrix ecosystem. yes.